fix(build): Make trackbase build possible without Geant3

karabowi opened the pull request:
Moved the add_subdirectory(trackbase) outside of if(Geant3_FOUND) condition.

Fixes issue #1609.

---

Checklist:

* [x] Followed the [Contributing Guidelines](https://github.com/FairRootGroup/FairRoot/blob/dev/CONTRIBUTING.md)

karabowi committed:
fix(build): Make trackbase build possible without Geant3

Moved the add_subdirectory(trackbase) outside of if(Geant3_FOUND) condition.

Fixes issue #1609.

ChristianTackeGSI reviewed (dismissed):
Generally this looks like the right thing to do.

If we want to support building without Geant3, do we have a testcase for this? (not sure, I haven't touched FairRoot for too long now).

P.S.: We probably should revive our CI one way or another?

karabowi committed:
feat(examples): Move `FairConstField` from `Tutorial3` to `common/passive`

The ConstField magnetic field is used by few examples: Tutorial1, Tutorial3, propagator,
By moving it to `common` the examples become more independent.

karabowi committed:
fix(Tutorial4): Fix the dependency on target `EventDisplay`

The build of `FairTutorialDet4PointDraw.cxx` dependend
on `opengl`, while it should depend on `TARGET FairRoot::EventDisplay`.
Fixed this.

karabowi committed:
fix(propagator):  Remove the dependency of the whole example on `EventDisplay`

Since only few classes really depend on `EventDisplay`, made only these few
really depend on the exsistance of `EventDisplay`.

Added `ExPassive` to link FairConstField.

karabowi committed:
fix(propagator): Remove the simulation macro dependency on Geant3

The macro depended on Geant3, because it uses `FairGeane` and `FairGeanePro`
inside an if function.
Added preprocessor directives to remove this dependency.

karabowi committed:
fix(examples): Move examples out of fake dependency, mainly on Geant3

Made order in the dependcies of many of the examples.
Most were placed in the `if(Geant3_FOUND)` which is not needed.
Moved `propagator` out of `if(TARGET FairRoot::EventDisplay`
following previous commit.

That increased the number of tests from 40 to 68 for the no-Geant3 case.

karabowi commented:
@ChristianTackeGSI . Agree on everything.
In principle the test of `trackbase` is the `propagator` example. 
It falsely depended on `EventDisplay`. Fixed that.
Cleaned a bunch of stuff.

karabowi commented:
@ChristianTackeGSI can you have a look at the PR again?

ChristianTackeGSI commented on examples/advanced/propagator/macros/runProp.C:
```diff
@@ -64,8 +64,10 @@ int runProp(std::string propName = "rk")
 
     // -----   Set propagator and run   ---------------------------------------
     if (propName == "geane") {
+#ifdef FAIRGeane_H
```
I really don't like it, if one matches on the include-guard of a header.

Either you include that header (then you know, you did), or you don't.

Please use some other preprocessor thing here.

karabowi replied:
@ChristianTackeGSI , done.

ChristianTackeGSI reviewed (changes requested):
Please remember to change the copyright year on all files that you changed.

ChristianTackeGSI reviewed (approved):
Assuming you were able to compile things / run tests…

hopefully our CI will come back sometime.

karabowi commented:
I have tested on Mac/Sep25 and Ubuntu/May25.
